modules/thermal_hydraulics/joss_paper.md

```diff
@@ -189,11 +189,44 @@ decreasing the number of input file iterations.
 
 # Documentation
 
-(Guillaume: refer to website)
+THM is extensively documented following the same object-oriented paradigm as the code base. Each
+object is accompanied by a documentation page that describes it, including its equations or a
+figure as relevant, that lists its parameters along with metadata about the parameters, that lists
+all the input files in the repository that use this object, and finally all the classes deriving from this object.
+
+Major groups of objects, usually derived from a single base class, are documented through the syntax documentation.
+Syntax documentation describes how to add a class of object to a simulation, and describes the common characteristics
+of this class of objects. Components and Closures are examples of a unique syntax in THM that also corresponds to
+base classes of groups of THM objects.
+
+This documentation page is hosted on the module website [@thm_website]. The website also notably hosts the software
+quality assurance (SQA) records, such as the testing requirement matrix or the failure analysis reports for example.
+The interested reader is referred to INL's SQA plan [@sqa] for more information.
 
 # Testing
 
-(Guillaume)
+The module relies on CIVET [@slaughter2021continuous] for continuous integration.
+Every pull request to the module runs the entire test suite for MOOSE, including the tests for THM.
+The test suite is comprehensive and aims to cover every feature available in the module. It
+consists of unit and regression tests, described below. The entire test suite is run with a wide variety of
+configurations, from compiling with the oldest and newest supported compilers, to running with several
+shared memory threads and distributed memory processes, on a variety of operating systems. The test suite
+is also run encapsulated in Valgrind to detect memory leaks. Every pull request to the module
+must include tests to cover any new features. This is ensured by the reviewer. The review process is detailed
+in INL's SQA plan [@sqa].
+
+Unit tests in THM are targeted at specific routines that can be accessed by creating the relevant object with
+example parameters. For example, creating a `Flux` object, we can check that the formulation of the numerical
+flux, whether with a centered scheme or with HLLC, are both consistent and symmetric.
+
+Regression tests are essentially created for each object to ensure that their output does not vary on a
+relevant test case. For example, each function or each auxiliary kernel is tested on a simple Cartesian mesh to make sure
+the field it produces is consistently the same. Components are often tested in the minimal configuration sufficient
+to satisfy the test requirement, to prove conservation of mass and momentum for example for a flow channel.
+
+Finally, the test suite includes numerous Jacobian tests to prove the correctness of the implementation of the
```

Comment:
I think we probably don't need to talk of Jacobian testing since it falls under "regression testing". However, that does remind me that it might be worth mentioning AD somewhere, but it's not too important I think.